InvenTree/company/serializers.py

```diff
@@ -95,8 +98,33 @@ class Meta:
             'notes',
             'parts_supplied',
             'parts_manufactured',
+            'remote_image',
         ]
 
+    def save(self):
+        """Save the Company instance"""
+        super().save()
+
+        company = self.instance
+
+        # Check if an image was downloaded from a remote URL
+        remote_img = getattr(self, 'remote_image_file', None)
+
+        if remote_img and company:
+            fmt = remote_img.format or 'PNG'
+            buffer = io.BytesIO()
+            remote_img.save(buffer, format=fmt)
+
+            # Construct a simplified name for the image
+            filename = f"company_{company.pk}_image.{fmt.lower()}"
+
+            company.image.save(
+                filename,
+                ContentFile(buffer.getvalue()),
+            )
+
+        return self.instance
```

Comment:
Question: Maybe a duplicate?

Reply:
Yes, a duplicate but consolidating this part only saves a few lines!